src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -30,24 +34,27 @@ import getPaymentModel from '../../../models/payment.server.model'
 import { getEncryptPendingSubmissionModel } from '../../../models/pending_submission.server.model'
 import { getEncryptSubmissionModel } from '../../../models/submission.server.model'
 import * as CaptchaMiddleware from '../../../services/captcha/captcha.middleware'
-import * as CaptchaService from '../../../services/captcha/captcha.service'
 import * as TurnstileMiddleware from '../../../services/turnstile/turnstile.middleware'
-import * as TurnstileService from '../../../services/turnstile/turnstile.service'
-import { createReqMeta, getRequestIp } from '../../../utils/request'
+import { Pipeline } from '../../../utils/pipeline-middleware'
+import { createReqMeta } from '../../../utils/request'
 import { getFormAfterPermissionChecks } from '../../auth/auth.service'
 import { MalformedParametersError } from '../../core/core.errors'
 import { ControllerHandler } from '../../core/core.types'
 import { setFormTags } from '../../datadog/datadog.utils'
 import { PermissionLevel } from '../../form/admin-form/admin-form.types'
 import * as FormService from '../../form/form.service'
-import { SGID_COOKIE_NAME } from '../../sgid/sgid.constants'
 import { SgidService } from '../../sgid/sgid.service'
 import { getOidcService } from '../../spcp/spcp.oidc.service'
 import { getPopulatedUserById } from '../../user/user.service'
 import * as VerifiedContentService from '../../verified-content/verified-content.service'
 import * as EncryptSubmissionMiddleware from '../encrypt-submission/encrypt-submission.middleware'
 import { reportSubmissionResponseTime } from '../submissions.statsd-client'
 
+import {
```

Comment:
TODO: write some notes on what is an ensure, the concept, and philosophy on this